crates/transpiler/src/passes/high_level_synthesis.rs

```diff
@@ -68,109 +68,172 @@ struct QubitTracker {
     ignored: Vec<bool>,
 }
 
-#[pymethods]
 impl QubitTracker {
-    #[new]
-    fn new(num_qubits: usize, qubits_initially_zero: bool) -> Self {
-        QubitTracker {
-            num_qubits,
-            state: vec![qubits_initially_zero; num_qubits],
-            enabled: vec![true; num_qubits],
-            ignored: vec![false; num_qubits],
-        }
-    }
-
-    fn num_qubits(&self) -> usize {
-        self.num_qubits
-    }
-
     /// Sets state of the given qubits to dirty
-    fn set_dirty(&mut self, qubits: Vec<usize>) {
+    fn set_dirty(&mut self, qubits: &[Qubit]) {
```

Comment:
_Strictly_ this could be as generic as `qubits: impl IntoIterator<Item: Borrow<Qubit>>`, but that's probably wild overkill.